Code/Engine/GameEngine/Gameplay/Implementation/GreyBoxComponent.cpp

```diff
@@ -159,7 +163,36 @@ void ezGreyBoxComponent::OnMsgExtractRenderData(ezMsgExtractRenderData& msg) con
       pRenderData->FillBatchIdAndSortingKey();
     }
 
-    msg.AddRenderData(pRenderData, ezDefaultRenderDataCategories::LitOpaque, ezRenderData::Caching::IfStatic);
+    // Determine render data category.
+    ezRenderData::Category category = m_CachedRenderDataCategory;
```

Comment:
Fix for greybox geometry not working with transparent materials

Reply:
A bit wonky since OnMsgExtractRenderData will be called from multiple threads but should work in this case.

---
In reply to: [400517954](https://github.com/ezEngine/ezEngine/pull/180#discussion_r400517954) [](ancestors = 400517954)